### PR TITLE
Metric data query alarms

### PR DESCRIPTION
**Pull request by eoinsha**
Fixes issue #3408

I added support for metric data queries in PutMetricAlarms sufficiently for my own needs. I'm not sure if the style in the implementation is ideal, but can modify as required.

- Add support for Metric Data Queries in `PutMetricAlarms` and `DescribeAlarms`
- Add unit test for `describe_alarms`

### CODE REVIEW

File: tests/test_cloudwatch/test_cloudwatch_boto3.py [hunk 1]
```diff
@@ -141,6 +141,72 @@ def test_describe_alarms_for_metric():
...
+        Metrics=metric_data_queries,
+        ComparisonOperator="GreaterThanThreshold",
+        Threshold=1.0,
+    )
+    alarms = conn.describe_alarms()
+    metric_alarms = alarms.get("MetricAlarms")
+    metric_alarms.should.have.length_of(2)
+    single_metric_alarm = [_ for _ in metric_alarms if "MetricName" in _][0]
+    multiple_metric_alarm = [_ for _ in metric_alarms if "MetricName" not in _][0]
+    single_metric_alarm["MetricName"].should.equal("cpu")
```
Comment (bblommers) [inline tests/test_cloudwatch/test_cloudwatch_boto3.py:206]: Can we extend the assertions here a bit? Would be good to verify that other properties are returned as expected, such as alarm.unit, metricstat, etc.

**Review by bblommers — CHANGES_REQUESTED** [review]

File: tests/test_cloudwatch/test_cloudwatch_boto3.py (hunk 1 above)
Comment (eoinsha) [reply tests/test_cloudwatch/test_cloudwatch_boto3.py:206]: @bblommers that makes sense. I'll revisit these tests and update the PR.

Comment (eoinsha) [reply tests/test_cloudwatch/test_cloudwatch_boto3.py:211]: @bblommers I added these assertions and fixed a missing field in DescribeAlarms that these assertions revealed!

**Review by bblommers — APPROVED** [review]
LGTM. Thanks @eoinsha!

**Comment by eoinsha** [comment]
Is it at all possible to have a PyPI release with this feature, @bblommers ?
Many thanks :-)

**Comment by bblommers** [comment]
Hi @eoinsha, we do have dev-releases, so this is definitely part of the latest release (1.3.17.dev230).
I don't have an ETA for a full release I'm afraid

**Comment by bblommers** [comment]
@eoinsha We now have a full release - moto 2.0.0. See #3722

**Comment by eoinsha** [comment]
> @eoinsha We now have a full release - moto 2.0.0. See #3722

This is awesome - thanks @bblommers 👏

**Pull request merged**